opendatahub-io/opendatahub-operator
# Implement template rendering action

## Description

This PR introduces a new rendering engine that is based on the go text/template package.

The template is rendered to the actual objects taking a subset of the `ReconciliationRequest` struct: `Instance` and `DSCI` so as an example, the `ServiceMeshMember` template that is part of the model registry component would looks like:

```yaml
apiVersion: maistra.io/v1
kind: ServiceMeshMember
metadata:
  name: default
  namespace: {{.Instance.Spec.RegistriesNamespace}}
spec:
  controlPlaneRef:
    namespace: {{ .DSCI.Spec.ServiceMesh.ControlPlane.Namespace }}
    name: {{ .DSCI.Spec.ServiceMesh.ControlPlane.Name }}

```


https://issues.redhat.com/browse/RHOAIENG-15534

## How Has This Been Tested?




1. Enable the model registry component in DSC
2. Check for the presence of a  `ServiceMeshMember` instance in the model registries namespace named `default` and that it contains the expected values
3. Check metrics related to the render action, it should include something like:
```
# HELP action_renderer_manifests_total Number of rendered resources
# TYPE action_renderer_manifests_total counter
action_renderer_manifests_total{controller="modelregistry",engine="kustomize"} 17
action_renderer_manifests_total{controller="modelregistry",engine="template"} 1
```


## Screenshot or short clip


## Merge criteria
